libraries/AP_RangeFinder/AP_RangeFinder_Lua.cpp

```diff
@@ -28,25 +28,40 @@ AP_RangeFinder_Lua::AP_RangeFinder_Lua(RangeFinder::RangeFinder_State &_state, A
 }
 
 
-// Set the distance based on a Lua Script
-bool AP_RangeFinder_Lua::handle_script_msg(float dist_m)
+// Process range finder data from a lua driver. The three wriable items are copyied from
+// state_arg to _state_pending. The update() method will copy this data into the backend
+// state structure. The get_state() method is called by a lau script to get the most recent
+// measurement.
+bool AP_RangeFinder_Lua::handle_script_msg(const RangeFinder::RangeFinder_State &state_arg)
 {
-    state.last_reading_ms = AP_HAL::millis();
-    _distance_m = dist_m;
+    WITH_SEMAPHORE(_sem);
+
+    _state_pending.last_reading_ms = AP_HAL::millis();
```

Comment:
In the past we have left the script to update this. That means that the script can cause the No Data state without having to just stop updating. 

That does requite that the script author fill it in correctly, but it gives more options.